Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/SurveyModule.kt

```diff
@@ -1,9 +1,62 @@
 package de.rki.coronawarnapp.datadonation.survey
 
+import android.content.Context
 import dagger.Module
+import dagger.Provides
+import dagger.Reusable
 import de.rki.coronawarnapp.datadonation.survey.consent.SurveyConsentModule
+import de.rki.coronawarnapp.datadonation.survey.server.DataDonationApiV1
+import de.rki.coronawarnapp.datadonation.survey.server.DataDonationHttpClient
+import de.rki.coronawarnapp.environment.datadonation.DataDonationCDNServerUrl
+import de.rki.coronawarnapp.http.HttpClientDefault
+import de.rki.coronawarnapp.http.RestrictedConnectionSpecs
+import de.rki.coronawarnapp.util.di.AppContext
+import okhttp3.Cache
+import okhttp3.ConnectionSpec
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.converter.protobuf.ProtoConverterFactory
+import java.io.File
+import javax.inject.Singleton
 
 @Module(
     includes = [SurveyConsentModule::class]
 )
-class SurveyModule
+class SurveyModule {
+
+    @Reusable
+    @DataDonationHttpClient
+    @Provides
+    fun cdnHttpClient(
+        @HttpClientDefault defaultHttpClient: OkHttpClient,
+        @RestrictedConnectionSpecs connectionSpecs: List<ConnectionSpec>
```

Comment:
Does the TechSpec ask for the restricted connection specs?

And if so, should we add it to the injected base `@DataDonationCDNHttpClient` as the survey and analytics use the same endpoint, they'll both need this?

Reply:
obsolete